cognite/client/_api/data_models.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import annotations
+
+from typing import Iterator, Sequence, cast, overload
+
+from cognite.client._api_client import APIClient
+from cognite.client._constants import DATA_MODEL_LIST_LIMIT_DEFAULT
+from cognite.client.data_classes.data_modeling.data_models import (
+    DataModel,
+    DataModelFilter,
+    DataModelList,
+)
+from cognite.client.data_classes.data_modeling.ids import DataModelId, VersionedDataModelingId
+from cognite.client.utils._identifier import DataModelingIdentifierSequence
+
+
+class DataModelsAPI(APIClient):
+    _RESOURCE_PATH = "/models/datamodels"
+
+    def __call__(
+        self,
+        chunk_size: int = None,
+        limit: int = None,
+        space: str | None = None,
+        inline_views: bool = False,
+        all_versions: bool = False,
+        include_global: bool = False,
+    ) -> Iterator[DataModel] | Iterator[DataModelList]:
+        """Iterate over data_models
+
+        Fetches data_models as they are iterated over, so you keep a limited number of data_models in memory.
+
+        Args:
+            chunk_size (int, optional): Number of data_models to return in each chunk. Defaults to yielding one data_model a time.
+            limit (int, optional): Maximum number of data_models to return. Default to return all items.
+            space: (str | None): The space to query.
+            inline_views (bool): Whether to expand the referenced views inline in the returned result.
+            all_versions (bool): Whether to return all versions. If false, only the newest version is returned,
+                                 which is determined based on the 'createdTime' field.
+            include_global (bool): Whether to include global views.
+
+        Yields:
+            Union[DataModel, DataModelList]: yields DataModel one by one if chunk_size is not specified, else DataModelList objects.
+        """
+        DataModelFilter(space, inline_views, all_versions, include_global)
```

Comment:
I assume this does some validation since the result isn't used? Perhaps add a comment.

Reply:
This was a mistake, fixed it.